frame/child-bounties/src/lib.rs

```diff
@@ -0,0 +1,892 @@
+	/// child-bounty curator.
+	CuratorProposed {
+		/// The assigned child-bounty curator of this bounty.
+		curator: AccountId,
+	},
+	/// The child-bounty is active and waiting to be awarded.
+	Active {
+		/// The curator of this child-bounty.
+		curator: AccountId,
+	},
+	/// The child-bounty is awarded and waiting to released after a delay.
+	PendingPayout {
+		/// The curator of this child-bounty.
+		curator: AccountId,
+		/// The beneficiary of the child-bounty.
+		beneficiary: AccountId,
+		/// When the child-bounty can be claimed.
+		unlock_at: BlockNumber,
+	},
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::generate_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config:
+		frame_system::Config + pallet_treasury::Config + pallet_bounties::Config
+	{
+		/// Maximum number of child-bounties that can be added to a parent bounty.
+		#[pallet::constant]
+		type MaxActiveChildBountyCount: Get<u32>;
+
+		/// Minimum value for a child-bounty.
+		#[pallet::constant]
+		type ChildBountyValueMinimum: Get<BalanceOf<Self>>;
+
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The parent bounty is not in active state.
+		ParentBountyNotActive,
+		/// The bounty balance is not enough to add new child-bounty.
+		InsufficientBountyBalance,
+		/// Number of child-bounties exceeds limit `MaxActiveChildBountyCount`.
+		TooManyChildBounties,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A child-bounty is added.
+		ChildBountyAdded { index: BountyIndex, child_index: BountyIndex },
+		/// A child-bounty is awarded to a beneficiary.
+		ChildBountyAwarded {
+			index: BountyIndex,
+			child_index: BountyIndex,
+			beneficiary: T::AccountId,
+		},
+		/// A child-bounty is claimed by beneficiary.
+		ChildBountyClaimed {
+			index: BountyIndex,
+			child_index: BountyIndex,
+			payout: BalanceOf<T>,
+			beneficiary: T::AccountId,
+		},
+		/// A child-bounty is cancelled.
+		ChildBountyCanceled { index: BountyIndex, child_index: BountyIndex },
+	}
+
+	/// Number of total child bounties.
+	#[pallet::storage]
+	#[pallet::getter(fn child_bounty_count)]
+	pub type ChildBountyCount<T: Config> = StorageValue<_, BountyIndex, ValueQuery>;
+
+	/// Number of child-bounties per parent bounty.
+	#[pallet::storage]
+	#[pallet::getter(fn parent_child_bounties)]
+	pub type ParentChildBounties<T: Config> =
+		StorageMap<_, Twox64Concat, BountyIndex, BountyIndex, ValueQuery>;
+
+	/// Child-bounties that have been added.
+	#[pallet::storage]
+	#[pallet::getter(fn child_bounties)]
+	pub type ChildBounties<T: Config> = StorageDoubleMap<
```

Comment:
@thiolliere this combined with `ChildBountyCount` is hinting at a `CountedDoubleMap`, can we do that?